### PR TITLE
check dashboard widget roles

### PR DESCRIPTION
**Pull request by Thiblef**

### CODE REVIEW

**Comment by ggrasset** [comment]
:+1:

**Comment by SOLenG** [comment]
:+1:

**Comment by itkg-alavieille** [comment]
not agree with this refacto

File: ApiBundle/Transformer/WidgetCollectionTransformer.php [hunk 1]
```diff
@@ -34,4 +40,22 @@ public function getName()
...
+    /**
+     * @param $widget
+     *
+     * @return null|string
+     */
+    protected function getWidgetRole($widget)
+    {
+        $role = null;
+
+        if (preg_match('/node/', $widget)) {
```
Comment (itkg-alavieille) [inline ApiBundle/Transformer/WidgetCollectionTransformer.php]: it's not extensible and it's work only with widget of node and content. an integrator can be add a custom widget so your code not works with her widget

File: ApiBundle/Transformer/WidgetCollectionTransformer.php [hunk 2]
```diff
@@ -21,7 +24,10 @@ public function transform($widgetCollection)
         $facade = new WidgetCollectionFacade();
 
         foreach ($widgetCollection as $widget) {
-            $facade->addWidget($this->getTransformer('widget')->transform($widget));
+            $role = $this->getWidgetRole($widget);
+            if ($this->authorizationChecker->isGranted($role)) {
+                $facade->addWidget($this->getTransformer('widget')->transform($widget));
+            }
```
Comment (itkg-ngilain) [inline ApiBundle/Transformer/WidgetCollectionTransformer.php]: Are you sure of this functionnality ? I thought the widgets were always displayed but there contents could be adjusted dependeing on the user.

**Comment by Thiblef** [comment]
@itkg-alavieille @nicolasThal @SOLenG @itkg-nanne

File: ApiBundle/Transformer/ContentCollectionTransformer.php [hunk 3]
```diff
@@ -23,7 +24,9 @@ public function transform($contentCollection, $contentType = null)
         $facade = new ContentCollectionFacade();
 
         foreach ($contentCollection as $content) {
-            $facade->addContent($this->getTransformer('content')->transform($content));
+            if ($this->authorizationChecker->isGranted(ContentTypeForContentPanelStrategy::ROLE_ACCESS_CONTENT_TYPE_FOR_CONTENT, $content)) {
```
Comment (itkg-alavieille) [inline ApiBundle/Transformer/ContentCollectionTransformer.php]: it's not better to just hide the button to edit content than not display the content himself.
Same for the node

File: ApiBundle/Transformer/WidgetCollectionTransformer.php [hunk 4]
```diff
@@ -11,8 +11,9 @@
  */
 class WidgetCollectionTransformer extends AbstractTransformer
 {
+
```
Comment (nicolasThal) [inline ApiBundle/Transformer/WidgetCollectionTransformer.php]: empty line to suppress

File: ApiBundle/Transformer/ContentCollectionTransformer.php (hunk 3 above)
Comment (Thiblef) [inline ApiBundle/Transformer/ContentCollectionTransformer.php]: Access role means the right to access it = see it, like in the left panel. I think it's not enough to only hide the edit button when you don't have the access role. The content should be hidden too.

**Comment by Thiblef** [comment]
@nicolasThal @itkg-alavieille

File: ApiBundle/Transformer/ContentTransformer.php [hunk 5]
```diff
@@ -75,7 +75,8 @@ public function transform($content)
             $facade->addAttribute($contentAttribute);
         }
 
-        if ($this->authorizationChecker->isGranted(ContentTypeForContentPanelStrategy::ROLE_ACCESS_UPDATE_CONTENT_TYPE_FOR_CONTENT)) {
+        if ($this->authorizationChecker->isGranted(ContentTypeForContentPanelStrategy::ROLE_ACCESS_CONTENT_TYPE_FOR_CONTENT)
+            && $this->authorizationChecker->isGranted(ContentTypeForContentPanelStrategy::ROLE_ACCESS_UPDATE_CONTENT_TYPE_FOR_CONTENT)) {
```
Comment (itkg-ngilain) [inline ApiBundle/Transformer/ContentTransformer.php]: Why check update AND access to edit ? I think it's the only location where we're doing this isn't it ? The ROLE_ACCESS_UPDATE_CONTENT_TYPE_FOR_CONTENT allows edition even if the user doen't have ROLE_ACCESS_CONTENT_TYPE_FOR_CONTENT no ?

File: ApiBundle/Transformer/NodeTransformer.php [hunk 6]
```diff
@@ -98,9 +98,11 @@ public function transform($node)
         $editionRole = $node->getNodeType() === NodeInterface::TYPE_TRANSVERSE? GeneralNodesPanelStrategy::ROLE_ACCESS_UPDATE_GENERAL_NODE:TreeNodesPanelStrategy::ROLE_ACCESS_UPDATE_NODE;
         $facade->editable = $this->authorizationChecker->isGranted($editionRole, $node);
 
-        $facade->addLink('_self_form', $this->generateRoute('open_orchestra_backoffice_node_form', array(
-            'id' => $node->getId(),
-        )));
+        if ($facade->editable && $this->authorizationChecker->isGranted(TreeNodesPanelStrategy::ROLE_ACCESS_TREE_NODE)) {
```
Comment (itkg-ngilain) [inline ApiBundle/Transformer/NodeTransformer.php]: ROLE_ACCESS_TREE_NODE or ROLE_UPDATE_TREE_NODE ?

File: ApiBundle/Transformer/ContentTransformer.php (hunk 5 above)
Comment (Thiblef) [inline ApiBundle/Transformer/ContentTransformer.php]: I check it to disable the edit button in the dashboard widget. The UBO should not see the edit button if he have no access role nor update role. But you're right, it's not the best idea I had. I'm gonna fix it in the view.

**Comment by Thiblef** [comment]
@itkg-ngilain @itkg-alavieille

**Comment by Thiblef** [comment]
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1285

**Pull request closed without merging**